tamr_unify_client/models/base_collection.py

```diff
@@ -69,3 +69,33 @@ def stream(self, resource_class):
 
     def __iter__(self):
         return self.stream()
+
+    @abstractmethod
+    def by_external_id(self, resource_class, external_id):
+        """Retrieve an item in this collection by relative ID.
+
+        Subclasses should override this method and pass in the specific
+        ``resource_class`` for that collection.
+
+        :param resource_class: Resource class corresponding to items in this collection.
+        :type resource_class: Python class
+        :param external_id: The external ID.
+        :type external_id: str
+        :returns: The specified item, if found.
+        :rtype: ``resource_class``
```

Comment:
should include a `:raises KeyError: If <condition when this exception is thrown>`

Reply:
added